pkg/manager/internal/controllers/servicecluster_controller.go

```diff
@@ -173,6 +177,15 @@ func (r *ServiceClusterReconciler) updateStatus(
 		corev1alpha1.ServiceClusterControllerReady)
 	serviceClusterReachable, _ := serviceCluster.Status.GetCondition(
 		corev1alpha1.ServiceClusterReachable)
+	timenow := metav1.Now()
+	if timenow.Sub(serviceClusterReachable.LastHeartbeatTime.Time) > r.MonitorGracePeriod {
+		serviceClusterReachable.Type = corev1alpha1.ServiceClusterReachable
+		serviceClusterReachable.LastTransitionTime = timenow
+		serviceClusterReachable.Status = corev1alpha1.ConditionUnknown
+		serviceClusterReachable.Message = fmt.Sprintf("cluster stopped posting heartbeats for at least %v", r.MonitorGracePeriod)
+		serviceClusterReachable.Reason = "GracePeriodTimeout"
+		serviceCluster.Status.SetCondition(serviceClusterReachable)
```

Comment:
```suggestion
		serviceCluster.Status.SetCondition(corev1alpha1.ServiceClusterCondition{
			Type:    corev1alpha1.ServiceClusterReachable,
			Status:  corev1alpha1.ConditionUnknown,
			Reason:  "GracePeriodTimeout",
			Message: fmt.Sprintf("Cluster stopped posting heartbeats for at least %v.", r.MonitorGracePeriod),
			// will only take affect, when the condition really changed
			// see code in .SetCondition
			LastTransitionTime: timenow,
		})
```

I think this makes it way easier to read and bring it in line with the other SetCondition calls.
The comment about LastTransitionTime hopefully clears up any confusion about us setting the property here.

Reply:
I did it the way I did because serviceClusterReachable is used down the line. I could reload the variable though

Reply:
done